test/Microsoft.OpenApi.Readers.Tests/V2Tests/OpenApiDocumentTests.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.OpenApi.Exceptions;
+using Microsoft.OpenApi.Models;
+using Microsoft.OpenApi.Readers.Exceptions;
+using Xunit;
+
+namespace Microsoft.OpenApi.Readers.Tests.V2Tests
+{
+    public class OpenApiDocumentTests
+    {
+        [Fact]
+        public void ShouldThrowWhenReferenceTypeIsInvalid()
+        {
+            var input = @"
+swagger: 2.0
+info: 
+  title: test
+  version: 1.0.0
+paths: 
+  '/':
+    get:
+      responses:
+        '200':
+          description: ok
+          schema:
+            $ref: '#/definition/does/notexist'
```

Comment:
>definition [](start = 21, length = 10)

nit, can we change this to something totally different, so that it's obvious it doesn't exist. At first glance, I was confused because I mistakenly read this as "definitions"?

Reply:
I accidentally found this scenario by mistyping :-)

---
In reply to: [194477506](https://github.com/Microsoft/OpenAPI.NET/pull/275#discussion_r194477506) [](ancestors = 194477506)